test/ecdsatest.c

```diff
@@ -57,343 +49,269 @@ static int restore_rand(void)
-
-        /* Store the two BIGNUMs in raw_buf. */
-        r_len = BN_num_bytes(sig_r);
-        s_len = BN_num_bytes(sig_s);
-        bn_len = (order + 7) / 8;
-        if (!TEST_false(r_len > bn_len)
-                || !TEST_false(s_len > bn_len))
-            goto builtin_err;
-        buf_len = 2 * bn_len;
-        if (!TEST_ptr(raw_buf = OPENSSL_zalloc(buf_len)))
-            goto builtin_err;
-        BN_bn2bin(sig_r, raw_buf + bn_len - r_len);
-        BN_bn2bin(sig_s, raw_buf + buf_len - s_len);
-
-        /* Modify a single byte in the buffer. */
-        offset = raw_buf[10] % buf_len;
-        dirt = raw_buf[11] ? raw_buf[11] : 1;
-        raw_buf[offset] ^= dirt;
-
-        /* Now read the BIGNUMs back in from raw_buf. */
-        if (!TEST_ptr(modified_sig = ECDSA_SIG_new()))
-            goto builtin_err;
-        if (!TEST_ptr(modified_r = BN_bin2bn(raw_buf, bn_len, NULL))
-                || !TEST_ptr(modified_s = BN_bin2bn(raw_buf + bn_len,
-                                                    bn_len, NULL))
-                || !TEST_true(ECDSA_SIG_set0(modified_sig,
-                                             modified_r, modified_s))) {
-            BN_free(modified_r);
-            BN_free(modified_s);
-            goto builtin_err;
-        }
-        sig_ptr2 = signature;
-        sig_len = i2d_ECDSA_SIG(modified_sig, &sig_ptr2);
-        if (!TEST_false(ECDSA_verify(0, digest, SHA512_DIGEST_LENGTH,
-                                     signature, sig_len, eckey)))
-            goto builtin_err;
-
-        /* Sanity check: undo the modification and verify signature. */
-        raw_buf[offset] ^= dirt;
-        if (!TEST_ptr(unmodified_r = BN_bin2bn(raw_buf, bn_len, NULL))
-                || !TEST_ptr(unmodified_s = BN_bin2bn(raw_buf + bn_len,
-                                                      bn_len, NULL))
-                || !TEST_true(ECDSA_SIG_set0(modified_sig, unmodified_r,
-                                             unmodified_s))) {
-            BN_free(unmodified_r);
-            BN_free(unmodified_s);
-            goto builtin_err;
-        }
-
-        sig_ptr2 = signature;
-        sig_len = i2d_ECDSA_SIG(modified_sig, &sig_ptr2);
-        if (!TEST_true(ECDSA_verify(0, digest, SHA512_DIGEST_LENGTH,
-                                    signature, sig_len, eckey)))
-            goto builtin_err;
-
-        /* cleanup */
-        ERR_clear_error();
-        OPENSSL_free(signature);
-        signature = NULL;
-        EC_KEY_free(eckey);
-        eckey = NULL;
-        EC_KEY_free(wrong_eckey);
-        wrong_eckey = NULL;
-        ECDSA_SIG_free(ecdsa_sig);
-        ecdsa_sig = NULL;
-        ECDSA_SIG_free(modified_sig);
-        modified_sig = NULL;
-        OPENSSL_free(raw_buf);
-        raw_buf = NULL;
+    /* skip built-in curves where ord(G) is not prime */
+    if (nid == NID_ipsec4 || nid == NID_ipsec3) {
+        TEST_info("ECDSA unsupported for curve %s", OBJ_nid2sn(nid));
```

Comment:
optional nit: maybe add an explicit `skipped: ` at the beginning of the message?